src/inmanta/deploy/scheduler.py

```diff
@@ -1121,7 +1126,6 @@ async def _update_scheduler_state_for_finished_deploy(
                 DeploymentResult.NEW,
             )
 
-            # TODO[#8453]: make sure that timer is running even if we return early
             if details.attribute_hash != attribute_hash or state.status is ComplianceStatus.UNDEFINED:
```

Comment:
@sanderr isn't the second part (`state.status is ComplianceStatus.UNDEFINED`) implied by the first in all practical cases?

Reply:
It would be, except for the thing where we don't enforce the hash diff. We emit a warning if we observe this, but that still doesn't prevent it. So it is, for the 99% case, but I wanted to cover the other case.
I seem to recall that Arnaud added this at some point as well, and I made the same remark you did, so it got reverted. It was only when I was working on it myself that I realized that it was actually a good safeguard.

Reply:
added to code